pkg/network/ebpf/c/runtime/usm.c

```diff
@@ -262,10 +262,17 @@ int uprobe__crypto_tls_Conn_Read__return(struct pt_regs *ctx) {
         return 0;
     }
 
-    char *buffer_ptr = (char*)call_data_ptr->b_data;
-    bpf_map_delete_elem(&go_tls_read_args, (go_tls_function_args_key_t*)&call_key);
+    char *buffer_ptr = (char *)call_data_ptr->b_data;
+    bpf_map_delete_elem(&go_tls_read_args, (go_tls_function_args_key_t *)&call_key);
+
+    // We need to flip the tuple to match the tuple from the write uprobe. This
+    // is needed for HTTP2 decoding to work properly, as without that tuple flip,
+    // we would use different dynamic tables for decoding the buffers from the read
+    // & write uprobes.
+    conn_tuple_t copy = *t;
+    flip_tuple(&copy);
```

Comment:
Isn't it a TLS hook, hence we want to avoid the tuple normalization?

Reply:
sorry it should have been removed